client/src/ui/BitbakeRecipesView.ts

```diff
@@ -29,10 +29,18 @@ class BitbakeRecipe extends vscode.TreeItem {
 class BitbakeTreeDataProvider implements vscode.TreeDataProvider<BitbakeRecipe> {
   readonly bitbakeWorkspace: BitbakeWorkspace
 
-  readonly onDidChangeTreeData: vscode.Event<BitbakeRecipe | undefined> = new vscode.EventEmitter<BitbakeRecipe | undefined>().event
+  private readonly _onDidChangeTreeData: vscode.EventEmitter<BitbakeRecipe | undefined> = new vscode.EventEmitter<BitbakeRecipe | undefined>()
+  readonly onDidChangeTreeData: vscode.Event<BitbakeRecipe | undefined> = this._onDidChangeTreeData.event
 
   constructor (bitbakeWorkspace: BitbakeWorkspace) {
     this.bitbakeWorkspace = bitbakeWorkspace
+
+    bitbakeWorkspace.onChange.on('recipeAdded', (recipe: string) => {
+      this._onDidChangeTreeData.fire(undefined)
```

Comment:
Why are we firing `undefine` here?

Reply:
Because the public event onDidChangeTreeData takes `BitbakeRecipeTreeItem | undefined`.
From this context, I don't have access to the recipe that's added or dropped. For optimization, a recipeTreeItem can be passed to only call getChildren() on it instead of the whole tree. However, add/drop recipe means I have to update the root node which corresponds to `undefined`. Likewise, the scanReady event requires to update all recipe nodes, so I also need to pass `undefined` for the root node.